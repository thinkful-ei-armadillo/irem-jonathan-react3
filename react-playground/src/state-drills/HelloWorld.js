import React, {Component} from 'react';


class HelloWorld extends Component {
    constructor(props){
        super(props);
        this.state={who: 'world'};
        
    this.handleWorldButton=()=>{this.setState({who: 'World!'});}
    this.handleFriendButton=()=>{this.setState({who: 'Friend!'});}
    this.handleReactButton=()=>{this.setState({who: 'React!'});}
}

    render(){
        return (
        <div className="Buttons">
            <p>Hello {this.state.who}</p>
            <button className="worldButton" onClick={this.handleWorldButton}>World</button>
            <button className="friendButton" onClick={this.handleFriendButton}>Friend</button>
            <button className="reactButton" onClick={this.handleReactButton}>React</button>
        </div>
        );
    }

}

export default HelloWorld;
