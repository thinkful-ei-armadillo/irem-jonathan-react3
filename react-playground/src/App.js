import React, { Component } from 'react';
import './App.css';
import HelloWorld from './state-drills/HelloWorld';
import Bomb from './state-drills/Bomb';


class App extends Component {
  static defaultProps = {
    state: {
      count: 1
    }
  }

  render() {
    return (
      <div>
        <div className="App">
          <HelloWorld who={HelloWorld.who}/>
        </div>
        <div>
          <Bomb />
        </div>
      </div>
    );
  }
}

export default App;
