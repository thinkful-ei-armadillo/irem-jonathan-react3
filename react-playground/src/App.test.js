import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';
import HelloWorld from './state-drills/HelloWorld';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe(`App Component`, () => {
  
  it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders empty given no seclection', () => {
  const wrapper = shallow(<App />)
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders with all tabs', () => {
  const wrapper = shallow(<App who={HelloWorld.who} bulletInChamber={8}/>)
  expect(toJson(wrapper)).toMatchSnapshot()
})
})