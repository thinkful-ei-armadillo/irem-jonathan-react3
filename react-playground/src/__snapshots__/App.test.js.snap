// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders empty given no seclection 1`] = `
<div>
  <div
    className="App"
  >
    <HelloWorld />
  </div>
  <div>
    <RouletteGun
      bulletInChamber={8}
    />
  </div>
  <div>
    <Accordion
      content="Lorem ipsum dolor sit amet consectetur adipisicing elit."
      key="0"
      title="Section 1"
    />
    <Accordion
      content="Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!"
      key="1"
      title="Section 2"
    />
    <Accordion
      content="Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?"
      key="2"
      title="Section 3"
    />
  </div>
</div>
`;

exports[`App Component renders with all tabs 1`] = `
<div>
  <div
    className="App"
  >
    <HelloWorld />
  </div>
  <div>
    <RouletteGun
      bulletInChamber={8}
    />
  </div>
  <div>
    <Accordion
      content="Lorem ipsum dolor sit amet consectetur adipisicing elit."
      key="0"
      title="Section 1"
    />
    <Accordion
      content="Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!"
      key="1"
      title="Section 2"
    />
    <Accordion
      content="Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?"
      key="2"
      title="Section 3"
    />
  </div>
</div>
`;
